### PR TITLE
Dbus job

### PR DESCRIPTION
**Pull request by ClydeByrdIII**
Following https://github.com/systemd/systemd/issues/20928#issuecomment-933733352, I've modified dbus-job.c to use the new macros.

From the comment

> After migrating the API, make sure to test everything still works as before. If you're not sure how to test the changes, open a PR for the changes and clearly mention that you're not sure how to test the changes and we'll make sure to help you out.

Unfortunately I'm not sure how to test this particular functionality aside from the unit tests.

cc @anitazha, @DaanDeMeyer

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
I'm pretty sure this is covered by CI. Is the first commit intended to be a part of this PR?

File: man/systemd.kill.xml [hunk 1]
```diff
@@ -84,9 +84,14 @@
         <para>Processes will first be terminated via <constant>SIGTERM</constant> (unless the signal to send
         is changed via <varname>KillSignal=</varname> or <varname>RestartKillSignal=</varname>). Optionally,
         this is immediately followed by a <constant>SIGHUP</constant> (if enabled with
-        <varname>SendSIGHUP=</varname>). If processes still remain after the main process of a unit has
-        exited or the delay configured via the <varname>TimeoutStopSec=</varname> has passed, the termination
-        request is repeated with the <constant>SIGKILL</constant> signal or the signal specified via
+        <varname>SendSIGHUP=</varname>). If processes still remain after:
+				<itemizedlist>
```
Comment (yuwata) [inline man/systemd.kill.xml]: Indentation?

**Comment by yuwata** [comment]
Looks mostly good, but one minor nit as commented in the above.

In the next time, please use the `git push --fource`, then you do not need to open a new PR.

**Comment by poettering** [comment]
Hmm, I am puzzled, but the first commit doesn't look right to me? It's news to me that KillMode=mixed would influence whether we wait for the main process or not. What made you think it does?

**Comment by ClydeByrdIII** [comment]
Sorry folks I'm a basically a git(hub) beginner after 5 years of mercurial.
@DaanDeMeyer  you're right; I wanted two separate PRs. I thought by making a separate branch I'd have made separated out my other change request. Without @anitazha baby sitting me, I guess I'm prone to mistakes.

I'll ask Anita for help before I attempt to split these back to two PRs.

> It's news to me that KillMode=mixed would influence whether we wait for the main process or not. What made you think it does?

@poettering  sorry, didn't want this discussion on this PR, but from my experience, the change is intending to make it clear that KillMode=control-group follows a different behavior from mixed..

From the behavior we have seen, mixed mode operates as described in the systemd.kill man page:

_KillSignal= is first sent to $MAINPID, after stopTimeout or $MAINPID exit, then FinalKillSignal= is sent to remaining processes._

We also depend on the observed behavior that $MAINPID exiting (independently of a stop job; e.g a crash) will lead to killing all other remaining processes in the cgroup.

This is all fine and expected. What was not clear was the behavior of control-group when $MAINPID exited.
Currently the man page may imply that killMode=control-mode behaves similarly to mixed, but it does not (on cgroup v2).

_KillSignal= is first sent to all processes, after stopTimeout, then FinalKillSignal= is sent to remaining processes._

Regardless of whether $MAINPID exited or not, systemd still waits stopTimeout for the remaining processes to react to killSignal. I expected FinalKillSignal to be sent as soon as $MAINPID exited and not stopTimeoutSecs later.

My change is an attempt to make it clear that control-group and mixed will result in FinalKillSignal being sent after a timeout to remaining processes in the cgroup, BUT only KillMode=mixed will have FinalKillSignal sent when $MAINPID exits.

I hope that makes sense and hope my change reflects that intent.

**Comment by poettering** [comment]
I think there's a misunderstanding somewhere. How precisely does your unit look like you tested this with? Most importantly, did you define an ExecStop=?

**Comment by anitazha** [comment]
Clyde removed the KillMode= change from here so let's continue the discussion on #22683. We can merge the dbus change independently.

**Comment by DaanDeMeyer** [comment]
> Sorry folks I'm a basically a git(hub) beginner after 5 years of mercurial.
@DaanDeMeyer you're right; I wanted two separate PRs. I thought by making a separate branch I'd have made separated out my other change request. Without @anitazha baby sitting me, I guess I'm prone to mistakes.

No worries, I miss Mercurial every day. Lack of PR stacks is a huge pain in github.

**Pull request merged**